src/amo/reducers/collections.js

```diff
@@ -357,15 +501,81 @@ export const createInternalCollection = ({
   slug: detail.slug,
 });
 
+type LoadCollectionIntoStateParams = {|
+  state: CollectionsState,
+  collection: ExternalCollectionDetail,
+  addons?: ExternalCollectionAddons,
+|};
+
+export const loadCollectionIntoState = (
+  { state, collection, addons }: LoadCollectionIntoStateParams
+): CollectionsState => {
+  const existingCollection = state.byId[collection.id];
+  const internalCollection = createInternalCollection({
+    detail: collection, items: addons,
+  });
+  // In case the new collection isn't loaded with add-ons,
```

Comment:
Sometimes we just load a collection and sometimes we load a collection + its add-ons. This is a separation that the API has made (see [listing collections](https://addons-server.readthedocs.io/en/latest/topics/api/collections.html#list) vs [listing add-ons](https://addons-server.readthedocs.io/en/latest/topics/api/collections.html#collection-add-ons-list)) because loading add-ons takes a big performance hit.

I would have designed the reducer a bit differently to make dealing with the two states easier but I don't think it's worth the effort to change it right now.